custom_components/resmed_myair/client/rest_client.py

```
@@ -656,10 +664,18 @@ async def get_user_device_data(self, initial: bool | None = False) -> Mapping[st
         )
         _LOGGER.debug("[get_user_device_data] records_dict: %s", redact_dict(records_dict))
         try:
-            device: Mapping[str, Any] = records_dict["data"]["getPatientWrapper"]["fgDevices"][0]
+            device: dict[str, Any] = records_dict["data"]["getPatientWrapper"]["fgDevices"][0]
         except Exception as e:
             _LOGGER.error("Error getting User Device Data. %s: %s", type(e).__name__, e)
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)